### PR TITLE
Parse docstrings as markdown

### PR DESCRIPTION
**Pull request by eiri**
## PR summary

We have markdown in docstrings of our generated modules. This causes issues with sphinx autodoc parser.

- Switch to autodoc2 and myst to properly parse markdown.
- Patch invalid markdown before parsing.
- Switch to "Read the Docs" theme and normalize TOC for better presentation.

Fixes: #78 

**Note: An existing issue is [required](https://github.com/IBM/cloudant-python-sdk/blob/main/CONTRIBUTING.md#PRs) before opening a PR.**

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [x] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by eiri** [comment]
Preview: https://ibm.github.io/cloudant-python-sdk/docs/dev/

_I'll rollback demo commit in `gh-pages after merge_

File: source/index.rst [hunk 1]
```diff
@@ -1,22 +1,30 @@
 .. Cloudant Python SDK documentation master file, created by
-   sphinx-quickstart on Tue Jan  5 13:47:50 2021.
-   Copyright IBM Corporation 2021.
+   sphinx-quickstart on Tue Jan  5 13:47:50 2021, 2023
```
Comment (eiri) [inline source/index.rst:2]:
```suggestion
   sphinx-quickstart on Tue Jan  5 13:47:50 2021.
```

File: source/index.rst [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
+   Copyright IBM Corporation 2021, 2023
```
Comment (eiri) [inline source/index.rst:3]:
```suggestion
   Copyright IBM Corporation 2021, 2023.
```

File: source/conf.py [hunk 3]
```diff
@@ -18,11 +18,10 @@
 sys.path.insert(0, os.path.abspath('..'))
 print(os.path.abspath('..'))
 
-
 # -- Project information -----------------------------------------------------
 
 project = 'Cloudant Python SDK'
-copyright = 'Copyright IBM Corp. 2021'
+copyright = 'Copyright IBM Corp. 2021, 2023'
```
Comment (eiri) [inline source/conf.py:24]:
```suggestion
copyright = 'Copyright IBM Corp. 2021, 2023.'
```

File: source/conf.py [hunk 4]
```diff
@@ -31,7 +30,33 @@
...
+
+autodoc2_packages = [
+    {
+        'path': '../ibmcloudant',
+        'exclude_files': [
+            'common.py',
+            'version.py',
+            'cloudant_base_service.py',
+            'couchdb_session_get_authenticator_patch.py'
+        ]
```
Comment (ricellis) [inline source/conf.py:48]: The session token manager is internal, so I think we can probably exclude it here as well.
```suggestion
            'couchdb_session_get_authenticator_patch.py',
            'couchdb_session_token_manager.py'
        ]
```

File: source/index.rst [hunk 5]
```diff
@@ -1,22 +1,30 @@
...
+-------
 
 .. toctree::
    :maxdepth: 4
-   :caption: Contents:
 
-   modules
+   cloudant_v1 <apidocs/ibmcloudant/ibmcloudant.cloudant_v1>
+   couchdb_session_authenticator <apidocs/ibmcloudant/ibmcloudant.couchdb_session_authenticator>
+   couchdb_session_token_manager <apidocs/ibmcloudant/ibmcloudant.couchdb_session_token_manager>
```
Comment (ricellis) [inline source/index.rst:24]: I think remove this one
```suggestion
```

File: source/index.rst [hunk 6]
```diff
@@ -1,22 +1,30 @@
...
 
-Welcome to Cloudant Python SDK's documentation!
-===============================================
+Cloudant Python SDK's Documentation
+===================================
+
+CloudantV1 class
+----------------
+
+:class:`CloudantV1 <ibmcloudant.cloudant_v1.CloudantV1>`
```
Comment (ricellis) [inline source/index.rst:14]: Really good idea to link this here.
Maybe in some later follow-on work we can figure out a way to add the available functions that correspond to API operations to the TOC or something.

File: requirements-docs.txt
Comment (ricellis) [inline requirements-docs.txt:1]: Any idea if dependabot will be able to pick this up? I know we have it set for `requirements` and `requirements-dev`, but I don't know offhand how flexible it is in that regard.

**Review by ricellis — COMMENTED** [review]
This is really nice, a few thoughts.

File: requirements-docs.txt
Comment (eiri) [reply requirements-docs.txt:1]: _presumably_ yes https://github.com/dependabot/dependabot-core/issues/3940#issuecomment-1227005267

File: source/index.rst (hunk 6 above)
Comment (eiri) [reply source/index.rst:14]: Yes, that'd be nice to have and it's trivial to add manual `cloudantv1.rst` with whatever content we want, but automatically getting to `toc` only section of another file is not obvious. Brute force hack would be to cut a middle out of generated `cloudant_v1.py` into a separate file and generate index from it, though I'm concerned what that'd do with links, so I didn't pursue that idea.

**Review by ricellis — APPROVED** [review]

**Pull request merged**